csharp/ql/src/semmle/code/csharp/Common.qll

```diff
@@ -0,0 +1,84 @@
+/**
+ * Provides classes and predicates for common functionality.
+ */
+import csharp
+
+/**
+ * Constructs a string containing the text `text` with a
+ * link to the element `e`.
+ */
+bindingset[text]
+string link(Element e, string text) {
+  exists(string path, int sl, int sc, int el, int ec, string url, Location location |
+    location = e.getLocation() and
+    location.hasLocationInfo(_, sl, sc, el, ec) and
+    path = location.getFile().getRelativePath() and
+    url = "relative://" + path + ":" + sl + ":" + sc + ":" + el + ":" + ec and
+    result = "[[\"" + text + "\"|\"" + url + "\"]]"
+  )
+}
+
+/**
+ * Holds if the object creation 'oc' is the creation of the reference type with the specified 'qualifiedName', or a class derived from 
+ * the class with the specified 'qualifiedName'.
+ */
+predicate isCreatingObject(ObjectCreation oc, string qualifiedName) {
+  exists(RefType t | t = oc.getType() |
+    t.getBaseClass*().hasQualifiedName(qualifiedName)
+  )
+}
+
+/**
+ * Holds if the method call 'mc' is returning the reference type with the specified 'qualifiedName. 
+ * and the target of the method call is a library method.
+ */
+predicate isReturningObject(MethodCall mc, string qualifiedName) {
+  mc.getTarget().fromLibrary() and 
+  exists (RefType t | t = mc.getType() |
+    t.hasQualifiedName(qualifiedName)
+  )
+}
+
+/**
+ * Holds if the method call 'mc' is a call on the library method target with the specified 'qualifiedName' and 'methodName', and an argument at 
+ * index 'argumentIndex' has the specified value 'argumentValue' (case-insensitive) 
+ */
+bindingset[argumentValue]
+predicate isMethodCalledWithArg(MethodCall mc, string qualifiedName, string methodName, int argumentIndex, string argumentValue) {
+  mc.getTarget().fromLibrary() and 
```

Comment:
This makes these methods quite hard to test, if you ever need to generate stubs for them. Can we just drop this requirement?

Reply:
Also for other places in this file.